### PR TITLE
Remove libc

### PR DESCRIPTION
**Pull request by lu-zero**
- [x] assert()
- [x] isprint()
- [ ] iswspace()
- [ ] iswalnum()

### CODE REVIEW

File: src/subtree.rs [hunk 1]
```diff
@@ -1643,9 +1581,7 @@ unsafe extern "C" fn ts_subtree__write_char_to_string(
         ) as size_t;
     } else if (0 as libc::c_int) < c
         && c < 128 as libc::c_int
-        && *(*__ctype_b_loc()).offset(c as isize) as libc::c_int
-            & _ISprint as libc::c_int as libc::c_ushort as libc::c_int
-            != 0
+        && ((c as u8).is_ascii_graphic() || c == ' ' as i32)
```
Comment (dodomorandi) [inline src/subtree.rs:1584]: Should we use a more generic whitespace checker?
```suggestion
        && ((c as u8).is_ascii_graphic() || (c as u8).is_ascii_whitespace())
```

**Review by dodomorandi — CHANGES_REQUESTED** [review]

File: src/subtree.rs (hunk 1 above)
Comment (lu-zero) [reply src/subtree.rs:1584]: Ideally you want to print `' '` for space and then use the hex value for the remaining cases not accounted above.

Comment (dodomorandi) [reply src/subtree.rs:1584]: My fault, didn't notice the previous cases!

**Review by dodomorandi — APPROVED** [review]

File: src/stack.rs [hunk 2]
```diff
@@ -2022,6 +1706,7 @@ pub unsafe extern "C" fn ts_stack_print_dot_graph(
         }
     }
     fprintf(f, b"}\n\x00" as *const u8 as *const libc::c_char);
+    libc::fclose(f);
```
Comment (dodomorandi) [inline src/stack.rs]: Here the problem: you have to close `f` only if you opened it with `fdopen`

**Review by dodomorandi — CHANGES_REQUESTED** [review]

File: src/stack.rs (hunk 2 above)
Comment (lu-zero) [reply src/stack.rs]: You are absolutely right!

Comment (lu-zero) [reply src/stack.rs]: Fixed, now the tests pass correctly on mac as well

**Review by dodomorandi — COMMENTED** [review]
Fine!

**Pull request merged**